spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -246,6 +183,65 @@ class CometSparkSessionExtensions
     }
   }
 
+  private def transformV2Scan(scanExec: BatchScanExec): SparkPlan = {
+
+    scanExec.scan match {
+      case scan: ParquetScan =>
+        val fallbackReasons = new ListBuffer[String]()
+        if (!CometBatchScanExec.isSchemaSupported(scan.readDataSchema)) {
+          fallbackReasons += s"Schema ${scan.readDataSchema} is not supported"
+        }
+
+        if (!CometBatchScanExec.isSchemaSupported(scan.readPartitionSchema)) {
+          fallbackReasons += s"Partition schema ${scan.readPartitionSchema} is not supported"
+        }
+
+        if (scan.pushedAggregate.nonEmpty) {
+          fallbackReasons += "Comet does not support pushed aggregate"
+        }
+
+        if (fallbackReasons.isEmpty) {
+          val cometScan = CometParquetScan(scanExec.scan.asInstanceOf[ParquetScan])
+          logInfo("Comet extension enabled for Scan")
+          CometBatchScanExec(
+            scanExec.copy(scan = cometScan),
+            runtimeFilters = scanExec.runtimeFilters)
+        } else {
+          withInfo(scanExec, withInfo(scanExec, fallbackReasons.mkString(", ")))
```

Comment:
same here

Reply:
How did this compile ??

Reply:
It compiled because it was calling this version of the method:

```scala
def withInfo[T <: TreeNode[_]](node: T, exprs: T*): T
```

Reply:
Right. I now remember writing that.